modules/dashboard/src/components/Admin/Admin.js

```diff
@@ -1,14 +1,122 @@
-import React from "react";
+import React, { useEffect } from "react";
 import Grid from '@material-ui/core/Grid';
 import { styled } from '@material-ui/core/styles';
+import { Contract, getDefaultProvider } from "ethers";
+import { getAddress, Interface, keccak256, solidityKeccak256 } from "ethers/utils";
 import { GetIncorrectProxyFactoryAddress, FixIncorrectProxyFactoryAddress } from "./IncorrectProxyFactory/IncorrectProxyFactoryAddressView";
+import { fromExtendedKey } from "ethers/utils/hdnode";
+import MinimumViableMultisig from "@connext/cf-funding-protocol-contracts/build/MinimumViableMultisig.json";
+import ProxyFactory from "@connext/cf-funding-protocol-contracts/build/ProxyFactory.json";
+import historicalData from "./historicalData";
 
 const RootGrid = styled(Grid)({
   flexGrow: 1,
   padding: "5%",
 });
 
+/*
+ * Notes re ProxyBytecode
+ * the first bytecode is "", this is important bc this signals that we should use the one from the factory
+ */
```

Comment:
can you elaborate? all of the proxy byte code on the hub side should be fixed except for these channels:

```0x1508eCF431F5DeF63d708fa657a9c8dB5d153a78
0xf6393A66a5A667D729b2b287117055466EDdDfBb
0x99A4c63cFf0Dcd50eeCF7B867C8a7aC89696A997
0xB559B1cad8eE3ea676fFC2eBA0a115D3BACD2b67
0x60EBf23291F0a0Ad51b01f997E37441217D24d7D```

Reply:
The scanner in this PR is more powerful than the nodes since it can check and confirm whether a channel is unrecoverable (eg if a toxic bytecode + mismatched factory was used to calculate a given multisig address).. I suspect that once we port it over to the node & re-run it we'll find that those 4 channels are unrecoverable.

Reply:
Note that this upgraded getCreate2Address will use a given (toxic) bytecode or default to fetching it from the factory if the given one is falsy (eg `""`)